src/PointwiseFunctions/AnalyticSolutions/Elasticity/AnalyticSolution.hpp

```diff
@@ -0,0 +1,61 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "PointwiseFunctions/AnalyticData/Elasticity/AnalyticData.hpp"
+#include "Utilities/FakeVirtual.hpp"
+#include "Utilities/Registration.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace Elasticity {
+/// Analytic solutions of the linear Elasticity equations
+namespace Solutions {
+
+/// Base class for analytic solutions of the linear Elasticity equations
+template <size_t Dim, typename Registrars>
+class AnalyticSolution
+    : public Elasticity::AnalyticData::AnalyticData<Dim, Registrars> {
+ private:
+  using Base = Elasticity::AnalyticData::AnalyticData<Dim, Registrars>;
+
+ protected:
+  /// \cond
+  AnalyticSolution() = default;
+  AnalyticSolution(const AnalyticSolution&) = default;
+  AnalyticSolution(AnalyticSolution&&) = default;
+  AnalyticSolution& operator=(const AnalyticSolution&) = default;
+  AnalyticSolution& operator=(AnalyticSolution&&) = default;
+  /// \endcond
+
+ public:
+  ~AnalyticSolution() override = default;
+
+  using registrars = Registrars;
+  using creatable_classes = Registration::registrants<registrars>;
+
+  /// \cond
+  explicit AnalyticSolution(CkMigrateMessage* m) noexcept : Base(m) {}
+  WRAPPED_PUPable_abstract(AnalyticSolution);  // NOLINT
+  /// \endcond
+
+  template <typename DataType, typename... RequestedTags>
+  tuples::TaggedTuple<RequestedTags...> variables(
+      const tnsr::I<DataType, Dim, Frame::Inertial>& x,
+      tmpl::list<RequestedTags...> /*meta*/) const noexcept {
+    return call_with_dynamic_type<
+        tuples::TaggedTuple<RequestedTags...>,
+        tmpl::filter<
+            creatable_classes,
+            std::is_base_of<AnalyticSolution<Dim, Registrars>, tmpl::_1>>>(
```

Comment:
1. Yes
2. No, because only a subset of the list of registrars are analytic solutions, and only for those can the `variables` function template be instantiated for these tags. On other words: The user can specify any analytic data or solution in the input file, but the analytic solution tags can only be retrieved from a subset of the creatable classes.